### PR TITLE
[None][refactor] BREAKING: Migrate LlmRequest fields to Python

### PR DESCRIPTION
**Pull request by Funatiq**
## Dev Engineer Review

- The changes affect only import formatting in `tensorrt_llm/_torch/pyexecutor/llm_request.py` and `tensorrt_llm/_torch/pyexecutor/py_executor.py`.
- No runtime behavior or API changes are shown.
- Review findings and test execution results were not supplied. Review severity counts are unavailable.

## QA Engineer Review

No test changes.

## Per-File QA Perspective

- `tensorrt_llm/_torch/pyexecutor/llm_request.py`: The change reformats a typing import. No observable behavior change is shown.
- `tensorrt_llm/_torch/pyexecutor/py_executor.py`: The change reformats imports, including multiline imports. No observable behavior change is shown.

## Description

* Migrate Python-consumed request payloads to pure Python.
  * `py_position_ids`, `py_guided_decoding_params`, `py_end_id`
* Make `py_multimodal_data` the canonical source and remove separate `mrope_rotary_cos_sin` and `mrope_position_deltas` fields.
* Use canonical `py_multimodal_data` and remove unused `multimodal_embedding`.
* fix: rename `_py_embedding_bias_1d` to `py_embedding_bias` in request and tests.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [ ] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/_torch/pyexecutor/py_executor.py [hunk 1]
```diff
@@ -5456,7 +5456,17 @@ def _forward_step_inter_pp(self,
         )
 
     def _validate_token_id_range(self, request: LlmRequest) -> None:
-        if isinstance(self.model_engine.model, DecoderModelForCausalLM):
+        model = self.model_engine.model
+        if not hasattr(model, "lm_head"):
+            return
+
+        num_embeddings = model.lm_head.num_embeddings
```
Comment (BowenFu) [inline tensorrt_llm/_torch/pyexecutor/py_executor.py]: Folding the `end_id` check in here is a nice way to drop the C++ `validate()` dependency. One thing I'm not sure about: this read used to sit inside the `DecoderModelForCausalLM` guard. Now any model with an `lm_head` reaches it, and as far as I can tell AutoDeploy's captured-graph wrapper proxies `lm_head` to a plain `nn.Linear`, which has no `num_embeddings`. That would raise `AttributeError` inside `_respond_if_invalid` and fail every AutoDeploy request. Would something like this keep those paths working?

```suggestion
        num_embeddings = getattr(model.lm_head, "num_embeddings", None)
        if num_embeddings is None:
            return
```

I think this is required for this PR unless AutoDeploy never exposes `lm_head` here. A quick AutoDeploy smoke run would settle it either way.

Comment (Funatiq) [reply tensorrt_llm/_torch/pyexecutor/py_executor.py]: Good catch. Codex also added `out_features` as an alternative to `num_embeddings`. See the new commit.
